src/app/space/[did]/root/[cid]/shard/[shard]/page.tsx

```diff
@@ -28,23 +29,38 @@ interface PageProps {
   }
 }
 
+const isCARLink = (link: Link): link is CARLink => link.code === CAR.codec.code
+
 export default function ItemPage ({ params }: PageProps): JSX.Element {
   const [{ client, spaces }] = useW3()
   const spaceDID = decodeURIComponent(params.did)
   const space = spaces.find(s => s.did() === spaceDID)
   const root = parseLink(params.cid)
-  const shard = parseLink(params.shard)
+  const shard = parseLink(params.shard).toV1()
 
   const storeKey = `/space/${spaceDID}/store/get?link=${shard}`
-  const store = useSWR<StoreGetSuccess|undefined>(storeKey, {
+  const store = useSWR<{ size: number } | undefined>(storeKey, {
     fetcher: async () => {
-      if (!client || !space) return
+      if (!client || !space || !(isCARLink(shard))) return
 
       if (client.currentSpace()?.did() !== space.did()) {
         await client.setCurrentSpace(space.did())
       }
 
-      return await client.capability.store.get(shard)
+      // First try to get the shard using the Blob protocol, then fall back to
+      // the Store protocol. Note that the Blob attempt can throw *or* return an
+      // error in the result.
+      return client.capability.blob.get(shard.multihash).then((result) => {
+        if (result.error) throw result.error
```

Comment:
Would be good to distinguish between a service error and a 404 here and only retry on 404.